src/runtime/pay-flow.js

```diff
@@ -50,6 +51,9 @@ export class PayStartFlow {
 
     /** @private @const {string} */
     this.sku_ = sku;
+
+    /** @private @const {!AnalyticsService} */
+    this.analyticsService_ = new AnalyticsService(deps);
```

Comment:
We shouldn't be creating the analytics service for each use. We should use a singleton. How about creating it in runtime and including it in deps?

Reply:
Done and a bunch of related to chagnes to that.